Automattic/jetpack
# VideoPress: Allow multiple values on videopress_rating and videopress_privacy_setting filters

Fixes #26827.

#### Changes proposed in this Pull Request:

* Changed the `videopress_rating` and `videopress_privacy_setting` to expect a list of CSV values and use an `IN` condition so we can filter for multiples values, just like other core endpoints work for filters like `author` or `slug`

#### Other information:


#### Jetpack product discussion



p1665681570531189-slack-C02TQF5VAJD.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* To filter videos using the `privacy_setting` value, request data in this way (you can use any combination of the `0`, `1` or `2` values):

```
curl -X GET -u "user:pass" "http://your-site.com/wp-json/wp/v2/media?mime_type=video/videopress&per_page=6&videopress_privacy_setting=0,1,2" | jq
```

* To filter videos using the `rating` value, request data in this way (you can use any combination of the `G`, `PG-13` or `R` values):

```
curl -X GET -u "user:pass" "http://your-site.com/wp-json/wp/v2/media?mime_type=video/videopress&per_page=6&videopress_rating=G,PG-13,R" | jq
```

* You can also combine the filters on the same request:

```
curl -X GET -u "user:pass" "http://your-site.com/wp-json/wp/v2/media?mime_type=video/videopress&per_page=6&videopress_rating=G,PG-13,R&videopress_privacy_setting=0,1,2" | jq
```

* After doing the request, confirm that only videos with the given `privacy_setting` or `rating` are returned
* To change the value for the `privacy_setting` or `rating`, do a request this way:

```
curl -X POST -u "user:pass" http://your-site.com/wp-json/wpcom/v2/videopress/meta -H 'Content-Type: application/json' -d '{"id": { ATTACHMENT ID },"rating": "PG-13","privacy_setting":"0" }' | jq
```

* You can also change the `privacy_setting` using the client:

![image](https://user-images.githubusercontent.com/6760046/195603806-8b05ea7c-5db8-4ae8-ac49-62af7a209335.png)
